### DISCUSSION TITLE
Webhook to trigger CI/CD pipeline when new prompt version is created

### DISCUSSION THREAD

**Discussion by tamdoEH** [langfuse/langfuse#1711] [Ideas]
### Describe the feature or potential improvement

Webhooks could be a great add-on that would attract more users. Whenever a prompt is deployed to production, it will trigger a CI/CD pipeline.

**Comment by marcklingen** [comment]
I like this idea as it is a creative use of webhooks. It could for example power the following workflow which helps overcome common limitations of playgrounds:

1. Use langfuse datasets to store a testing dataset
2. Create a new prompt version
3. Trigger a runner in CI via a Webhook that runs the application (pulling the latest prompt) in a loop on the dataset and pushed scores to langfuse
4. See evaluation metrics for prompt versions as langfuse dataset runs

**Comment by jannikmaierhoefer** [comment]
Hi all, we added webhooks for prompt changes (creation, update, deletion) which allows you to trigger a runner in CI via a Webhook that runs the application (pulling the latest prompt) in a loop on the dataset and pushes scores to langfuse: https://langfuse.com/docs/prompt-management/features/webhooks-slack-integrations

**Comment by jachinsaikiasonowal** [comment]
We ran into the exact same bottleneck when running multi-agent swarms in production.

Whenever an agent hallucinates in step 3 of a workflow, waiting 45 minutes for a full git commit -> container build -> backend redeploy just to change 10 words in a system prompt kills velocity.

We ended up building an open-source tool called PromptMatrix (https://promptmatrix.github.io) around 3 rules:

1. Decoupled in-memory serving: Prompts are served via a <5ms local cache endpoint (`pm.serve()`), so hot-swaps don't introduce HTTP latency bottlenecks into synchronous agent loops.
2. CLI/CI Eval Gating: You can run deterministic, rule-based evals in your CI/CD pipeline (`pmx eval --type rule_based`) to block promotions if the prompt breaks schema conformance.
3. 1-Click Rollbacks: If a prompt causes regressions, you roll back the registry version instantly without git reverts.

The core is 100% MIT open-source (SQLite local floor + Python SDK). Curious how your team is balancing latency vs eval gating in CI right now?
